File: mslibrary-kotlindocs/src/main/java/kr/msleague/msgui/gui/api/event/MSGuiCloseEvent.java
package kr.msleague.msgui.gui.api.event;

import kr.msleague.msgui.gui.MSGui;
import lombok.Getter;
import lombok.Setter;
import org.bukkit.entity.Player;
import org.bukkit.event.HandlerList;

@Getter
@Setter
public class MSGuiCloseEvent extends MSGuiEvent {
    private static HandlerList handlerList = new HandlerList();
    private long time;
    public MSGuiCloseEvent(Player player, MSGui<?> gui, long time) {
        super(player, gui);
        this.time = time;
    }

    public static HandlerList getHandlerList(){
        return handlerList;
    }

    @Override
    public HandlerList getHandlers() {
        return handlerList;
    }

}

File: mslibrary-gui/huni-gui/src/main/java/kr/msleague/msgui/api/events/MSGuiCloseEvent.kt
package kr.msleague.msgui.api.events

import kr.msleague.msgui.api.interfaces.MSGui
import kr.msleague.msgui.api.interfaces.MSGuiViewer
import org.bukkit.event.Cancellable

class MSGuiCloseEvent(
    viewer: MSGuiViewer,
    gui: MSGui,
): MSGuiEvent(viewer, gui), Cancellable {

    private var cancelled: Boolean = false
    override fun isCancelled(): Boolean = cancelled
    override fun setCancelled(cancel: Boolean) { cancelled = cancel }

}